src/main/java/com/bullhornsdk/data/api/helper/RestApiSession.java

```diff
@@ -85,7 +85,12 @@ public RestApiSession(BullhornRestCredentials bullhornRestCredentials) {
 		this.restCredentials = bullhornRestCredentials;
 		this.restTemplate = RestTemplateFactory.getInstance();
 		this.dateTimeBhRestTokenWillExpire = getNow();
-		createSession();
+		if(bullhornRestCredentials.credentialCheck()) {
```

Comment:
minor nitpick - need to match formatting of repo - there needs to be a space after the word if